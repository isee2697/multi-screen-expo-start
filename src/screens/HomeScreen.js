import React from 'react';
import PropTypes from 'prop-types';
import { ScrollView, Text, View } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import { gStyle } from '../constants';

// components
import Touch from '../components/Touch';

const HomeScreen = ({ navigation }) => (
  <SafeAreaView style={gStyle.container}>
    <ScrollView contentContainerStyle={gStyle.contentContainer}>
      <Text style={gStyle.text}>Home Content Area</Text>

      <View style={gStyle.spacer80} />

      <Touch
        onPress={() => navigation.navigate('MultiBase')}
        text="jump to Multi tab"
      />
    </ScrollView>
  </SafeAreaView>
);

HomeScreen.navigationOptions = {
  header: null
};

HomeScreen.propTypes = {
  // required
  navigation: PropTypes.object.isRequired
};

export default HomeScreen;
