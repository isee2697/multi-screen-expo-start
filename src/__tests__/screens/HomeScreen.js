import React from 'react';
import { shallow } from 'enzyme';

import HomeScreen from '../../screens/HomeScreen';

const navigation = { navigate: jest.fn() };
const screen = shallow(<HomeScreen navigation={navigation} />);

describe('<HomeScreen />', () => {
  it('renders correctly', () => {
    expect(screen).toMatchSnapshot();
  });

  it('onPress', () => {
    screen
      .find('Touch')
      .props()
      .onPress();
  });
});
