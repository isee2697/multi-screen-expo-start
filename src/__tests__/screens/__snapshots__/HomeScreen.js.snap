// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<withOrientation
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingTop": 30,
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#1d2d3a",
        }
      }
    >
      Home Content Area
    </Component>
    <Component
      style={
        Object {
          "height": 80,
          "width": "100%",
        }
      }
    />
    <Touch
      accesCType="button"
      accesLabel={null}
      accesTrait="button"
      accessible={true}
      icon={null}
      onPress={[Function]}
      opacityActive={0.8}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1d2d3a",
          "borderColor": "#1d2d3a",
          "borderRadius": 4,
          "borderWidth": 0.5,
          "height": 48,
          "justifyContent": "center",
          "paddingHorizontal": 24,
          "paddingVertical": 8,
        }
      }
      text="jump to Multi tab"
      textStyle={
        Object {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    />
  </ScrollViewMock>
</withOrientation>
`;
